doc/architectural/compilation-and-development.md

```diff
@@ -48,35 +111,31 @@ contained in the `doc/` directory and gives some options for these
 tools. All of these could be improved and patches are very welcome. In
 some cases the algorithms used are described in the relevant papers.
 
-## Accessing doxygen documentation ##
-
 The doxygen documentation can be [accessed online](http://cprover.diffblue.com).
-To build it locally, run `doxygen` in `/src`. HTML output will be created in
-`/doc/html`. The index page is `/doc/html/index.html`.
-
-## Coding standards ##
-
-See <a
-href="https://github.com/diffblue/cbmc/blob/master/CODING_STANDARD.md">
-`CODING_STANDARD.md`</a> file in the root of the CBMC repository.
-
-CPROVER is written in a fairly minimalist subset of C++; templates and
-meta-programming are avoided except where necessary.
-External library dependencies are avoided (only STL and a SAT solver
-are required). Boost is not used. The `util` directory contains many
-utilities that are not (yet) in the standard library.
-
-Patches should be formatted so that code is indented with two space
-characters, not tab and wrapped to 80 columns. Headers for doxygen
-should be given (and preferably filled!) and the author will be the
-person who first created the file. Add doxygen comments to
-undocumented functions as you touch them. Coding standards
-and doxygen comments are enforced by CI before a patch can be
-merged by running `clang-format` and `cpplint`.
-
-Identifiers should be lower case with underscores to separate words.
-Types (classes, structures and typedefs) names must end with a `t`.
-Types that model types (i.e. C types in the program that is being
-interpreted) are named with `_typet`. For example `ui_message_handlert`
-rather than `UI_message_handlert` or `UIMessageHandler` and
-`union_typet`.
\ No newline at end of file
+To build it locally, run `doxygen` in `src/`. HTML output will be created in
+`doc/html/`. The index page is `doc/html/index.html`.
+
+
+\section compilation-and-development-section-formatting Formatting
+
+The <a
+href="https://github.com/diffblue/cbmc/blob/develop/CODING_STANDARD.md">
+`CODING_STANDARD.md`</a> file in the root of the CBMC repository contains
+guidance on how to write code for the CBMC repository. This includes
+which language features can be used and formatting rules.
+
+C++ code can be automatically reformatted in the correct way by running
+`clang-format`. There are more details in
+[CODING_STANDARD.md](https://github.com/diffblue/cbmc/blob/develop/CODING_STANDARD.md#using-clang-format).
+
+
+\section compilation-and-development-section-linting Linting
+
+There is also a linting script, `scripts/cpplint.py`. There is a wrapper
+script to run `cpplint.py` only on lines that have been changed from the
+base branch:
+
+    scripts/run_lint.sh <base-branch>
```

Comment:
I'd give an example base-branch just because I don't think its meaning is necessarily well understood.